### ISSUE TITLE
Privacy provider user delete error if no sessions

### ISSUE THREAD

**Issue by ericmerrill** [danmarsden/moodle-mod_attendance#677]
If attendance is present in a course, but has no sessions in it, you get an error from get_in_or_equal():

```
Debugging: Coding error detected, it must be fixed by a programmer: moodle_database::get_in_or_equal() does not accept empty arrays in 
* line 357 of /mod/attendance/classes/privacy/provider.php: call to moodle_database->get_in_or_equal()
* line 215 of /mod/attendance/classes/privacy/provider.php: call to mod_attendance\privacy\provider::delete_user_from_session_attendance_log()
* line 229 of /admin/tool/wp/classes/course_reset_api.php: call to mod_attendance\privacy\provider::delete_data_for_user()
...
```
I've truncated the full trace stack because it is being called from custom code, but even so, there is a clear error in the implementation of delete_data_for_user(). Since if no session IDs are found, that empty array is passed to get_in_or_equal, which doesn't work (without a special param, but the results may be unexpected).

**Comment by danmarsden** [comment]
Thanks Eric, I've pushed a fix for that into the 4.0 and higher supported branches. (interesting to see it installed on what appears to be a WP site though... I haven't got round to checking / adding proper support for WP myself but would be interested to see a PR if you were planning to do the work on your end! :-)

**Comment by ericmerrill** [comment]
@danmarsden Thanks for the quick turnaround on this!
I'll poke it a bit on the WP site, but I expect that does exactly what I expected.

Basically in WP, there is a feature where individual user progress in a course can be reset, and for plugins that don't have explicit support (which only included plugins do), it uses that built in privacy function to do it if available.
